### ISSUE TITLE
The "Network" tab should provide visibility into non-HttpClient HTTP traffic

### ISSUE THREAD

**Issue by brianquinlan** [flutter/devtools#4829]
The "Network" tab doesn't provide any visibility into HTTP traffic sent through clients other than HttpClient e.g. `package:cupertino_http`, `package:cronet_http`.

**Comment by kenzieschmoll** [comment]
@CoderDake @bkonyi could be similar to the issues with package:dio support

**Comment by bkonyi** [comment]
I'm not sure about `package:dio` (it might be using `HttpClient` internally), but `package:cupertino_http` and `package:cronet_http` are FFI packages that hook into some native libraries.

**Comment by hanskokx** [comment]
I experienced the same thing. Here's a super simple piece of code to demonstrate:
![image](https://user-images.githubusercontent.com/1911919/209997505-f6eafc90-ef04-45c1-986f-1657cd01acc3.png)

**Comment by brianquinlan** [comment]
@kenzieschmoll : `package:dio` uses `HttpClient` internally so I would expect it's traffic to be visible in the Network tab - isn't it?

@hanskokx : The default `Client` in `package:http` also uses `HttpClient` for it's implementation (except on web) so I would also expect it's traffic to be visible in the Network tab. Are you using a different `Client`?

**Comment by fennelhans** [comment]
@brianquinlan Nope. Just the default. There's nothing particularly exciting about the code or environment, either. About as simple of an app as you can write, running on Windows.

**Comment by brianquinlan** [comment]
Do I understand correctly that the Networking tab is driven from data collected in [`HttpProfiler`](https://github.com/dart-lang/sdk/blob/main/sdk/lib/_http/http_impl.dart#L7)

and then consumed by [`network_profiling.dart`](https://github.com/dart-lang/sdk/blob/main/sdk/lib/io/network_profiling.dart) using a pull model?

**Comment by CoderDake** [comment]
@brianquinlan yes those are the basics of it :)
I have some fixes coming through the pipe that may end up fixing some of the issues you are seeing.
There was an issue with request id's overflowing in devtools so requests were sometimes not showing up at all.


I've been updating this issue when I make progress:
https://github.com/flutter/devtools/issues/2860

If the issue you are seeing is that requests are not showing up in the inspector, it may be worth trying this workaround: https://github.com/flutter/devtools/issues/5194#issuecomment-1419408378

Running Devtools on desktop avoids that overflow error so you may then be able to see the requests showing up in the list.
(unfortunately the response tab will still be a bit buggy for now though)

**Comment by brianquinlan** [comment]
How about we move `HttpProfiler` (and the supporting classes) to `dart:developer`? 

To be useful to 3p HTTP client implementations, we'd have to make generic versions of some methods e.g. [`formatConnectionInfo`](https://github.com/dart-lang/sdk/blob/main/sdk/lib/_http/http_impl.dart#L107).

**Comment by bkonyi** [comment]
Moving this functionality to `dart:developer` makes sense to me, but we'll want to do some design work beforehand since `HttpProfiler` was only written with `dart:_http` in mind.

**Comment by brianquinlan** [comment]
It seems like the classes in `vm_service` would be a good starting point e.g. [`HttpProfileRequest`](https://api.flutter.dev/flutter/vm_service/HttpProfileRequest-class.html).

The classes in `dart:developer` would have to be mutable though. And have different names to avoid confusion - though these names are the obvious ones.

**Comment by mirabo-vuong-nt** [comment]
any update for this issue?

**Comment by brianquinlan** [comment]
There is a design discussion [here](https://docs.google.com/document/d/1kedr5KENFZyg5aEHBXvBPMtjlpM4YAoWqDtWxiOuW00/edit?resourcekey=0-ZENajKyQulzEmi4I1oP53Q) - feedback welcome!

**Comment by srawlins** [comment]
This is implemented. The cupertino_http, cronet_http, and ok_http packages all use the [http_profile](https://pub.dev/packages/http_profile) package to track HTTP requests, and report them, via the VM service, to DevTools.

There were also some timing issues fixed by @bkonyi, which will be available in Dart 3.8, and should be available on the Dart / Flutter beta channels now.

If you experience "missing requests" in the Network screen in the latest Flutter beta (3.31 I believe) or the latest Dart beta, please file a new issue.
